tests/lib/prepare-restore.sh

```diff
@@ -136,6 +136,46 @@ build_rpm() {
     fi
 }
 
+build_arch_pkg() {
+    base_version="$(head -1 debian/changelog | awk -F '[()]' '{print $2}')"
+    version="1337.$base_version"
+    packaging_path=packaging/arch
+    archive_name=snapd-$version.tar
+
+    rm -rf /tmp/pkg
+    mkdir -p "/tmp/pkg/sources/snapd"
+    cp -ra -- * "/tmp/pkg/sources/snapd/"
+
+    # shellcheck disable=SC2086
+    (tar -C /tmp/pkg/sources -cf "/tmp/pkg/$archive_name" "snapd")
+    cp "$packaging_path"/* "/tmp/pkg"
+
+    # fixup PKGBUILD which builds a package named snapd-git with dynamic version
+    #  - update pkgname to use snapd
+    #  - kill dynamic version
+    #  - packaging functions are named package_<pkgname>(), update it to package_snapd()
+    #  - update source path to point to local archive instead of git
+    #  - fix package version to $version
+    sed -i \
+        -e "s/^source=.*/source=(\"$archive_name\")/" \
+        -e "s/pkgname=snapd.*/pkgname=snapd/" \
+        -e "s/pkgver=.*/pkgver=$version/" \
+        -e "s/package_snapd-git()/package_snapd()/" \
+        /tmp/pkg/PKGBUILD
+    awk '
```

Comment:
Hm, is this really the only way? It looks really hard to read, what are the inputs/outputs here?

Reply:
We need to comment out this whole `pkgver` section: https://github.com/bboozzoo/snapd/blob/a3a463199cc12b4dbb70e92339f7c45c518317eb/packaging/arch/PKGBUILD#L33 then
update `source=("git+ssh://...")` to point to tar.gz, rename `package_snapd-git()` to `package_snapd()` and rename the pacakge to `snapd`.